### DISCUSSION TITLE
Rustdesk Server Key mandatory?

### DISCUSSION THREAD

**Discussion by Craftoncu** [rustdesk/rustdesk-server#464] [Q&A]
In RustDesk, only the initiating PC requires the key, while the remote PC does not. How does this impact encryption?

If the initiating PC has a key but the remote PC doesn't, is the entire remote control communication still fully encrypted?

**Comment by xlionjuan** [comment]
You need on both sides

**Comment by klixx23** [comment]
I don't think that's true.

I have installed it as described in the following instructions:
[https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/ubuntu-server/docker/](https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/ubuntu-server/docker/)


The remote client (Client A) only has the RustDesk server stored in the settings under Network > ID/Relay Server > ID Server. 

The initiating PC (Client B) has stored the RustDesk server and the public key from the server in the settings under Network > ID/Relay Server > ID Server.

The following is possible and also encrypted:
- Connection from Client B to Client A works
- Connection from Client A to Client B does not work. Error message: Key mismatch

I have also tested the following:
-  Both clients only have the RustDesk stored, but no public key -> Connection failed: Error message: Key mismatch

The current situation is that only the initiating PC requires the public key, but the PC seeking help only requires the host or IP address of the RustDesk server.

I have tested it with software versions:
-  iOS App: 1.3.6
- Mac OS and Windows App: 1.3.7
- RustDesk server: 1.1.14

Currently only one client needs the public key to establish a connection.
However, this may be a bug or it may be intentional that RustDesk works this way. I also assumed that both clients need the public key from the server
